File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
    let difference;
    let j = numbers.length - 1;
    for (let i = 0; i < j ; i++) {
        difference = target - numbers[i];
        while (numbers[j-1] >= difference && j-1 > i) {
            j--;
        }
        if (numbers[i] + numbers[j] === target) {
            return [i+1, j+1];
        }
    }
};
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # create a number:index mapping
        mymap = {}
        for index, number in enumerate(nums):
            if target - number in mymap:
                return [mymap[target - number], index]
            else:
                mymap[number] = index
            

